### ISSUE TITLE
Url Shortening Service

### ISSUE THREAD

**Issue by npathai** [npathai/system-design-with-code#1]
- [X] From long url to short url
- [ ] From short to long url
    - [X] Basic conversion
    - [ ] Who should append the hostname in short url
    - [ ] Considerations for load balancing
    - [ ] What if there are multiple hostnames
- [X] Redirect the user to long url when short url is visited
- [ ] Url expiry
- [ ] User Authentication
    - [ ] Basic authentication or OpenId Connect
    - [ ] Authenticated User will get Analytics and Tracking
    - [ ] Without authentication create url with default expiry maybe a week
    - [ ] Delete url

**Comment by IAmPramod** [comment]
@npathai We can think of logging ip of the users visiting any short url. This will help in gathering stats and creating reports.

**Comment by npathai** [comment]
@IAmPramod Good idea. I will consider that. Currently I am converting both microservices to Micronaut framework.

I started quick and dirty but now when I thought about discovery dashboard, graphana and Prometheus, zipkin tracing, health checks, etc. It would have required hell lot of manual work to get everything going.

So I have taken decision to move to Micronaut. Will push update soon.

**Comment by npathai** [comment]
@IAmPramod Hard work of porting both services to Micronaut is complete. Phew! Now adding new features should not be that much pain. Check it out. I will update the diagram to reflect the changes. In terms of starting the application, it remains the same.

**Comment by IAmPramod** [comment]
@npathai Sure, will pull the code and go through it.
